### PR TITLE
fix(reset-tools/reset_media.dart): fix exceptions on unmounting

### PR DESCRIPTION
**Pull request by medicalwei**
We received a report that, if using a disk without partition table:

```
sudo dd if=/dev/zero of=/dev/sda bs=512 count=1
sudo mkfs.fat /dev/sda
```

is it is possible to trigger error that USB media remains unformatted.

However, after my investigation I also found that partition unmounted error is not handled properly.  This merge request handles both `org.freedesktop.UDisks2.Error.NotMounted` error and treat it as nothing happened, as well as properly unmounting USB media without partition table.

I am waiting for colleague's response before actual merge.

### CODE REVIEW

**Comment by medicalwei** [comment]
I got response from @cyruslien that this PR fixed his use case.

**Review by d-loose — APPROVED** [review]
LGTM :+1:

@spydon is off this week, not sure if you want/need to wait for his approval as well.

**Comment by medicalwei** [comment]
This change does not change the UI (flutter) side of things, and I think it is okay for him to comment on the code and improve it later.

**Pull request merged**
